micrometer-core/src/test/java/io/micrometer/core/instrument/binder/logging/Log4j2MetricsTest.java

```diff
@@ -144,4 +144,20 @@ void noDuplicateLoggingCountWhenMultipleNonAdditiveLoggersShareConfig() {
         logger1.info("Hello, world!");
         assertThat(registry.get("log4j2.events").tags("level", "info").counter().count()).isEqualTo(1);
     }
+
+    @Issue("#2176")
+    @Test
+    void asyncLogShouldNotBeDuplicated() throws IOException {
+        ConfigurationSource source = new ConfigurationSource(getClass().getResourceAsStream("/binder/logging/log4j2-async-logger.xml"));
+        Configurator.initialize(null, source);
+
+        Logger logger = LogManager.getLogger(Log4j2MetricsTest.class);
+
+        new Log4j2Metrics().bindTo(registry);
+
+        assertThat(registry.get("log4j2.events").tags("level", "info").counter().count()).isEqualTo(0);
+        logger.info("Hello, world!");
+        assertThat(registry.get("log4j2.events").tags("level", "info").counter().count()).isEqualTo(1);
```

Comment:
I think we have a race here since the logger is asynchronous, there's no guarantee that the counter will be incremented by this point.

Reply:
I still wonder about the theoretical race, but after much debugging, it seems like the issue was probably due to test execution order and global state from before the execution of the newly added test. Changing the cleanup method to `@BeforeEach` seems to have fixed it consistently for me locally.